packages/input/SliderInput.tsx

```diff
@@ -12,19 +13,29 @@ import { SliderInputProps } from './types'
 function SliderInput({
   value,
   onChange,
+  onLabelClick,
   min = 0,
   max = 100,
   step = 1,
   minLabel,
   maxLabel,
   getLabel = (val) => String(val),
+  borderNone,
+  labels,
 }: SliderInputProps): React.ReactElement {
   const fillPercentage = ((value - min) / (max - min)) * 100
+  const LabelComponent = onLabelClick ? LabelButton : Label
+  const createClickHandler = (value: number) => {
```

Comment:
Shouldn't this be wrapped in useCallback?

Reply:
It's complicated to add useCallback and keep the rules of hooks. I would appreciate it if you suggest a solution to that problem e.g. with "add suggestion" button)

Reply:
mmm, why won't this work?
```
cont handleClick = useCallback(value => onLabelClick?.(value), [onLabelClick])
```

Reply:
It won't work because the button clickHandler doesn't contain a value, so you need to generate clickHandler with a bound value. Also, we can't wrap every handler with a bound value with `useCallback` because the order and amount of `useCallback` calls should remain the same for every render. Still, the options array can be changed during the component lifecycle, which will break the "rules of hooks" and be very complicated to debug.

Reply:
@karinamaulitova can you show the issue in this sample? https://stackblitz.com/edit/react-playground-practice-qwmzyk?file=sample.jsx

Reply:
The main point of adding useCallback is to avoid rerenderings caused by changing of callback link. If I get it right in your example, `onClick` handlers will be created on every render anyway. And there is no way to avoid it since we need to bind values.

Reply:
hmm, actually makes sense, yeah, no huge difference, even may make it worse, you are right 👍 

Btw can we at least simplify the call via optional chaining?

```
    if (onLabelClick) {
      return () => onLabelClick(value)
    }
    return undefined
```